accepted/0000-timezones.md

```diff
@@ -0,0 +1,102 @@
+
+These rules should define a consistent way of display dates on the UI and make this information as clear as possible for the user.
+
+The user should be able to easily understand in which timezone a date is displayed and aware if there is any difference from the timezone
+he is accessing from.
+
+# Glossary
+[glossary]: #glossary
+
+When talking about uyuni there are 4 different Timezones contexts that are important to understand and distinguish:
+
+* *Managed systems timezone:* the timezone of each minion/client. This doesn't have any impact on this RFC, but it's important to mention.
+
+* *Uyuni Server timezone:* the configured timezone in the server where uyuni is deployed 
+
+* *User preferences timezone:* the defined timezone on the user preferences settings. Here we should be able to choose the timezone, date format (yyyyddmm) and the time format (12/24 hours).
+
+* *Client browser timezone:* the browser/pc timezone where the client is accessing from. This timezone shouldn’t affect any displayed date, but the user should be aware of it.
+
+
+# Detailed design
+[design]: #detailed-design
+
+There are 3 topics that we need to have in consideration here:
+
+## Display of dates
+
+The display of dates should always take in consideration the 3 user preferences settings: timezone, date-format and time-format.
+
+**For example:** '2019-09-10 13:00:00 **CEST**'
+
+At the moment only the setting timezone is implemented. The property date-format and time-format could be added on a second iteration, 
+but the code should be prepared to easily support them.
+
+Example:
+
+![timezones](images/timezones2.png)
+
+Ideally, these dates should be formatted server side using the already existing java methods:
+
+```java
+LocalizationService.getInstance().formatCustomDate(date)
+```
+
+For the new stack (spark+react) if you want to convert a JSON to use the correct format, use the following GSON date adapter `UserLocalizationDateAdapter` 
+that will convert your class dates to the user defined format.
+
+The previous gson adapter 'ECMAScriptDateAdapter' that converts dates to UTC should be avoided.
+
+## Manipulation of dates on the browser
+
+Since the browser Date implementation doesn't support timezones, if we ever need to do any date manipulation more than displaying a date
+ we should never use the object Date() directly but the `moment-timezone` library.
+
+This will require a refactor to the component react component `DateTimePicker` to properly manage timezones using the library `moment-timezone`
+instead of manually calculating the diff between the user preference and browser timezone.
+
+## Create awareness to the user regarding the used timezones
+
+The user should always be aware in which timezone his account is configured and from where he is accessing from.
+
+To create context to the user regarding timezones, 2 clocks should be added to the top. One with the user configured timezone and
```

Comment:
While I like the idea of the two different clocks, I am not sure I like the content of them. At the moment we could have three different timezone values set:
- browser
- user
- server

What has the most important impact is the difference between the client side and the server side: this difference will be reflected in scheduling actions. I'd better present the 2 clocks only if the server timezone is different that the one that is shown in the UI. And the one in the UI should always be in line with the user preference setting. To me the browser one should be completely overridden (and described clearly in docs). Unless, I'd be even fine with showing three different clocks: browser time, user time, server time. UI design can be really non intrusive adding small icons and showing the clocks only on a popup box. By this, user is aware of everything.

Reply:
humm I am not sure I understood everything here. 

The important time here is the "user settings" one and not the server one. Scheduling actions always assume the user settings one, the server time is irrelevant. So, the only important clocks differences are the browser (local user timezone) and user settings (the one defined in the account settings)

> I'd better present the 2 clocks only if the server timezone is different than the one that is shown in the UI

Agreed, I will add this to the rfc